// Code generated by mockery v2.39.1. DO NOT EDIT.

package mocks

import mock "github.com/stretchr/testify/mock"

// FnWithError is an autogenerated mock type for the FnWithError type
type FnWithError struct {
	mock.Mock
}

// Execute provides a mock function with given fields:
func (_m *FnWithError) Execute() error {
	ret := _m.Called()

	if len(ret) == 0 {
		panic("no return value specified for Execute")
	}

	var r0 error
	if rf, ok := ret.Get(0).(func() error); ok {
		r0 = rf()
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// NewFnWithError creates a new instance of FnWithError. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewFnWithError(t interface {
	mock.TestingT
	Cleanup(func())
}) *FnWithError {
	mock := &FnWithError{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
